ChromeExtension/js/serviceWorker.js

```diff
@@ -0,0 +1,129 @@
+chrome.runtime.onInstalled.addListener(async function (details) {
```

Comment:
most of the functions here are moved from the background script `background.js` since we need to be using service worker now in manifest v3

https://developer.chrome.com/docs/extensions/develop/migrate/to-service-workers#differences-with-sws